modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/portlet/shared/search/CPOptionFacetsPortletSharedSearchContributor.java

```diff
@@ -97,19 +98,22 @@ public void contribute(
 				frequencyThreshold = GetterUtil.getInteger(
 					portletPreferences.getValue("frequencyThreshold", null),
 					frequencyThreshold);
+				maxOptions = GetterUtil.getInteger(
+					portletPreferences.getValue("maxOptions", null),
+					maxOptions);
 				maxTerms = GetterUtil.getInteger(
 					portletPreferences.getValue("maxTerms", null), maxTerms);
 			}
 
 			serializableFacet.setFacetConfiguration(
 				buildFacetConfiguration(
 					serializableFacet.getFieldName(), frequencyThreshold,
-					maxTerms));
+					maxOptions));
```

Comment:
This looks like the cause of the impedibug. Did you test it before sending the PR?

Reply:
@andrea-ale-sbarra Yes, i did test it, and it's working correctly. There's two facet configurations because there's the [Top Facets](https://github.com/liferay-commerce/liferay-portal/pull/4495/files/f6da529ba9a532757e131d6c8fc5e3c2e8883ef7#diff-7abf7cc526dd9b788c5561738eaefdd0523286d2cc38c6c1cc2349ae8244b7d4R103-R106) (Options/Specifications), and [Sub-Facets](https://github.com/liferay-commerce/liferay-portal/pull/4495/files/f6da529ba9a532757e131d6c8fc5e3c2e8883ef7#diff-7abf7cc526dd9b788c5561738eaefdd0523286d2cc38c6c1cc2349ae8244b7d4R126-R128) (Each Option/Specification)
The Top Facets configuration needs to use the variables maxSpecifications/maxOptions introduced by Brian, and the Sub-Facets need to use maxTerms

What was causing the impedibug was that Brian forgot to use the new variables (maxOptions/maxSpecifications) as the number of max Sub-facets. That means that maxTerms was being used as the number of max Top-Facets and max Sub-facets, so if it was set to 2, there would be 2 Specifications fetched, as well as 2 terms for each one.

That's why the solution was just a change of variables, if we use the maxOptions/maxSpecifications as the actual number for the maximum number of Options/Specifications to be fetched, the fix works correctly.